alpakaConfig.cmake

```diff
@@ -253,47 +253,76 @@ IF(ALPAKA_ACC_GPU_CUDA_ENABLE)
                 SET(ALPAKA_CUDA_ARCH sm_20 CACHE STRING "Set GPU architecture" FORCE)
             ENDIF(IS_CUDA_ARCH_UNSUPPORTED)
 
-            LIST(APPEND CUDA_NVCC_FLAGS "-arch=${ALPAKA_CUDA_ARCH}")
+            SET(ALPAKA_CUDA_COMPILER "nvcc" CACHE STRING "CUDA compiler")
+            SET_PROPERTY(CACHE ALPAKA_CUDA_COMPILER PROPERTY STRINGS "nvcc;clang")
 
-            IF(NOT MSVC)
-                LIST(APPEND CUDA_NVCC_FLAGS "-std=c++11")
-                SET(CUDA_HOST_COMPILER "${CMAKE_CXX_COMPILER}")
-            ENDIF()
+            OPTION(ALPAKA_CUDA_FAST_MATH "Enable fast-math" ON)
+            OPTION(ALPAKA_CUDA_SHOW_REGISTER "Show kernel registers and create PTX" OFF)
+            OPTION(ALPAKA_CUDA_KEEP_FILES "Keep all intermediate files that are generated during internal compilation steps (folder: nvcc_tmp)" OFF)
 
-            IF(CMAKE_BUILD_TYPE MATCHES "Debug")
-                LIST(APPEND CUDA_NVCC_FLAGS "-g" "-G")
-            ENDIF()
+            IF(ALPAKA_CUDA_COMPILER MATCHES "clang")
+                LIST(APPEND _ALPAKA_COMPILE_OPTIONS_PUBLIC "--cuda-gpu-arch=${ALPAKA_CUDA_ARCH}")
 
-            OPTION(ALPAKA_CUDA_FAST_MATH "Enable fast-math" ON)
-            IF(ALPAKA_CUDA_FAST_MATH)
-                LIST(APPEND CUDA_NVCC_FLAGS "--use_fast_math")
-            ENDIF()
+                # This flag silences the warning produced by the Dummy.cpp files:
+                # clang: warning: argument unused during compilation: '--cuda-gpu-arch=sm_20'
+                LIST(APPEND _ALPAKA_COMPILE_OPTIONS_PUBLIC "-Qunused-arguments")
```

Comment:
This is not the problem. Clang warns for all sm's. The problem is that the Dummy.cpp file does not contain any CUDA code and therefore the flag is useless for the compilation of this file.